sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2553,8 +2553,9 @@ class Analyzer(override val catalogManager: CatalogManager)
               // a table `t` has two columns `c1` and `c2`, for query `SELECT ... FROM t
               // GROUP BY c1 HAVING c2 = 0`, even though we can resolve column `c2` here, we
               // should undo it later and fail with "Column c2 not found".
-              agg.child.resolve(u.nameParts, resolver).map(TempResolvedColumn(_, u.nameParts))
-                .getOrElse(u)
+              agg.child.resolve(u.nameParts, resolver).map(_.transformUp {
+                case a: Attribute => TempResolvedColumn(a, u.nameParts)
```

Comment:
shall we update `TempResolvedColumn` to only take `Attribute`?

Reply:
Here actually we shouldn't transform attributes, because for nested fields this `u.nameParts` will not correspond to the attribute. Updated to specifically match Alias.